sopel/plugins/handlers.py

```diff
@@ -609,26 +613,28 @@ def get_version(self) -> Optional[str]:
 
         if version is None and hasattr(self.module, "__package__"):
             try:
-                version = importlib_metadata.version(self.module.__package__)
+                version = importlib.metadata.version(self.module.__package__)
```

Comment:
Just noticed that this creates a mypy error now:

> sopel/plugins/handlers.py:616: error: Argument 1 to "version" has incompatible type "str | None"; expected "str"  [arg-type

It would make mypy happy if we ensure that `self.module.__package__ is not None` first, but if there's a less verbose way to spell type-safety here I'd be happy to use that instead:

```py
        if (
            version is None
            and hasattr(self.module, "__package__")
            and self.module.__package__ is not None
        ):
```

(I'd expect the simpler check `getattr(self.module, "__package__", None) is not None` to achieve the same effect, but mypy doesn't interpret that as ensuring that `self.module.__package__` isn't None.)

Reply:
Two other options that occur to me:

1) drop the `hasattr()` check, catch the `AttributeError` in the handler
2) spell the guard condition as `if version is None and version := getattr(self.module, "__package__"):`

I don't really like option (2) (it reeks of cleverness) but (1) seems like it should be fine. Not that there's anything wrong with your suggestion of an additional clause in the conditional.

Reply:
Option (2) assigns the package name to the variable named `version`, but why? Maybe it "reeks of cleverness" because it's trying to be _too_ clever? 😁

Option (1) doesn't silence mypy; it emits the same error with:

```diff
+        if version is None:
             try:
                 version = importlib.metadata.version(self.module.__package__)
+            except AttributeError:
+                # module probably doesn't have a __package__; skip
+                pass
```

Reply:
oh, hrm, right. I think the Right Way™ to do it is the additional condition, then.